addons/pinniped/config-controller/utils/utils.go

```diff
@@ -5,27 +5,50 @@
 package utils
 
 import (
+	"context"
 	"fmt"
+	"reflect"
+	"strings"
 
 	"github.com/go-logr/logr"
+	k8serror "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
 	corev1 "k8s.io/api/core/v1"
 	clusterapiv1beta1 "sigs.k8s.io/cluster-api/api/v1beta1"
-	"sigs.k8s.io/controller-runtime/pkg/client"
-	"sigs.k8s.io/controller-runtime/pkg/event"
-	"sigs.k8s.io/controller-runtime/pkg/predicate"
 
 	"github.com/vmware-tanzu/tanzu-framework/addons/pinniped/config-controller/constants"
 	tkgconstants "github.com/vmware-tanzu/tanzu-framework/pkg/v1/tkg/constants"
 )
 
-// IsAddonType returns true if the secret is type `tkg.tanzu.vmware.com/addon`
-func IsAddonType(secret *corev1.Secret) bool {
-	return secret.Type == constants.TKGAddonType
+// IsClusterBootstrapType returns true if the secret is type `tkg.tanzu.vmware.com/addon`
+func IsClusterBootstrapType(secret *corev1.Secret) bool {
+	return secret.Type == constants.ClusterBootstrapManagedSecret
 }
 
-// HasAddonLabel returns true if the `tkg.tanzu.vmware.com/addon` label matches the parameter we pass in
-func HasAddonLabel(secret *corev1.Secret, label string) bool {
-	return secret.Labels[constants.TKGAddonLabel] == label
+// ContainsPackageName returns true if the `tkg.tanzu.vmware.com/package-name` label contains the package name we pass in
+func ContainsPackageName(secret *corev1.Secret, expectedName string) bool {
+	name, labelExists := secret.Labels[constants.PackageNameLabel]
+	return labelExists && strings.Contains(name, expectedName)
+}
+
+func GetClusterNameFromSecret(secret *corev1.Secret) (string, error) {
+	for _, ownerRef := range secret.OwnerReferences {
```

Comment:
Both OwnerRef and the `ClusterNameLabel` should be there since they get added by the [clusterbootstrap controller](https://github.com/vmware-tanzu/tanzu-framework/blob/main/addons/controllers/clusterbootstrap_controller.go#L1096-L1111).  They also utilize both [when mapping secrets to clusters](https://github.com/vmware-tanzu/tanzu-framework/blob/main/addons/controllers/clusterbootstrap_handlers.go#L97-L116).  Looking more at their description, it seems `OwnerRef` would be preferred in our case since the `Type` can be matched, but at this point in the code (we're in the reconcile phase), we have already filtered down to the secrets we care about and are just trying to find the cluster that matches that secret... so I figured why not try to get the cluster name both ways?  But I'm open.